src/main/scala/mesosphere/marathon/core/launchqueue/LaunchStats.scala

```diff
@@ -146,18 +149,24 @@ object LaunchStats extends StrictLogging {
     * @param backOffUntil timestamp until which no further launch attempts will be made
     */
   case class QueuedInstanceInfoWithStatistics(
-      runSpec: RunSpec,
-      inProgress: Boolean,
-      instancesLeftToLaunch: Int,
-      finalInstanceCount: Int,
-      backOffUntil: Option[Timestamp],
-      startedAt: Timestamp,
-      rejectSummaryLastOffers: Map[NoOfferMatchReason, Int],
-      rejectSummaryLaunchAttempt: Map[NoOfferMatchReason, Int],
-      processedOffersCount: Int,
-      unusedOffersCount: Int,
-      lastMatch: Option[OfferMatchResult.Match],
-      lastNoMatch: Option[OfferMatchResult.NoMatch],
-      lastNoMatches: Seq[OfferMatchResult.NoMatch]
-  )
+    runSpec: RunSpec,
+    inProgress: Boolean,
+    instancesLeftToLaunch: Int,
+    finalInstanceCount: Int,
+    backOffUntil: Option[Timestamp],
+    startedAt: Timestamp,
+    rejectSummaryLastOffers: Map[NoOfferMatchReason, Int],
+    rejectSummaryLaunchAttempt: Map[NoOfferMatchReason, Int],
+    processedOffersCount: Int,
+    unusedOffersCount: Int,
+    lastMatch: Option[OfferMatchResult.Match],
+    lastNoMatch: Option[OfferMatchResult.NoMatch],
+    lastNoMatches: Seq[OfferMatchResult.NoMatch]
+  ) {
+    def queueDelay(clock: Clock) : Option[QueueDelay] = {
```

Comment:
Why is this an Option if it always returns Some?

Reply:
I was adhering to previous semantics and overlooked this. Should have caught this during refactor, will fix this. Good catch.